src/main/java/core/basesyntax/service/impl/operation/ReturnOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+
+public class ReturnOperation implements OperationHandler {
+    private final FruitStorageDao storageDao;
+
+    public ReturnOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(String fruitName, Integer quantity) {
+        Integer previousQuantity;
+        try {
+            previousQuantity = storageDao.getQuantity(fruitName);
+        } catch (RuntimeException e) {
+            previousQuantity = 0;
```

Comment:
Catching a `RuntimeException` to handle the case where a fruit is not present in the storage is a bit broad. Consider catching a more specific exception if possible, or checking for the fruit's existence before attempting to get its quantity to avoid using exceptions for control flow.